scripts/utils/retry-func.sh

```diff
@@ -0,0 +1,28 @@
+retry() {
+  local status
+  local interval=1
+  local retry=0
+  local -r factor=2
+  local -r max_retries=10
+  while true; do
+    "$@" 2> /tmp/errors.txt && break
+    cat /tmp/errors.txt
+    status=$?
+    ((retry += 1))
+    if [ $retry -gt $max_retries ]; then
+      echo "error: Command failed and reached max retries" 1>&2
+      return $status
+    fi
```

Comment:
The `-gt` conflicts with the error message.

Let's take `max_retries` as 1. With `-gt`, the input command will run twice.

Reply:
changed -gt to -ge

Reply:
But now `max_retries=10` is wrong, it will be a total of 10 attempts but only 9 of them will be retries :smile: 

Not sure what's the intended behavior. 10 total attempts? Or 11?

Reply:
10 total is fine

ok I've renamed max_retries => max_tries  ....